topology/connection_creator_impl.h

```diff
@@ -218,36 +218,39 @@ ConnectionCreator::target_driven_connect_( Layer< D >& source,
 #pragma omp parallel // default(none) shared(source, target, masked_layer,
                      // target_begin, target_end)
   {
+    const int thread_id = kernel().vp_manager.get_thread_id();
+
     for ( std::vector< Node* >::const_iterator tgt_it = target_begin;
           tgt_it != target_end;
           ++tgt_it )
     {
-      const thread target_thread = ( *tgt_it )->get_thread();
+      Node* const tgt =
+        kernel().node_manager.get_node( ( *tgt_it )->get_gid(), thread_id );
+      const thread target_thread = tgt->get_thread();
 
-      if ( target_thread != kernel().vp_manager.get_thread_id() )
+      // check whether the target is on our thread
+      if ( thread_id != target_thread )
+      {
         continue;
+      }
 
       if ( target_filter_.select_model()
-        && ( ( *tgt_it )->get_model_id() != target_filter_.model ) )
+        && ( tgt->get_model_id() != target_filter_.model ) )
         continue;
 
       const Position< D > target_pos =
-        target.get_position( ( *tgt_it )->get_subnet_index() );
+        target.get_position( tgt->get_subnet_index() );
 
       if ( mask_.valid() )
         connect_to_target_( pool.masked_begin( target_pos ),
           pool.masked_end(),
-          *tgt_it,
+          tgt,
           target_pos,
-          target_thread,
+          thread_id,
           source );
       else
```

Comment:
since the connect_to_target call is getting kind of multi-line, adding { } to the 'if' would ease the reading a lot